content/docs/nextjs/github-public-repo.md

```diff
@@ -0,0 +1,292 @@
+    </button>
+  )
+}
+```
+
+> _Note:_ For brevity, the example above configures many steps in a single file, but **a few components can be configured in different places**. For example you could put the `EditLink` in a Layout component, or setup the Github Provider only on certain pages.
+
+Now that **\_app.tsx** is ready. Let's setup up the backend API.
+
+## API functions
+
+We will need a few API functions to handle GitHub authentication and [Preview Mode](https://nextjs.org/docs/advanced-features/preview-mode). With Next.js, any functions in the `pages/api` directory are are mapped to `/api/*` endpoints.
+
+We've created a few [scripts](https://github.com/dwalkr/hygen-next-tinacms-github) to **help generate the required files**. From the console, run:
+
+```bash
+$ npx hygen-add https://github.com/dwalkr/hygen-next-tinacms-github
+$ npx hygen next-tinacms-github bootstrap --format ts
+```
+
+_Note:_ if your **pages directory is not in the root**, you will need to supply a `--dir [subDir]` option for this last script:
+
+```bash
+$ npx hygen next-tinacms-github bootstrap --format ts --dir src
+```
+
+You should see a few API functions have been setup in your project:
+
+- `preview.ts`: Contains API function to enter [Preview Mode](https://nextjs.org/docs/advanced-features/preview-mode), and set the preview-data with content stored in the cookies.
+- `proxy-github.ts`: Contains API function to attach the user's auth token, and proxy requests to the GitHub API.
+- `create-github-auth-token.ts`: Helper for creating a `createCreateAccessToken` server function.
+
+## Auth Redirects
+
+We will also need a page to redirect the user to while authenticating with GitHub. This component will render as a modal during authentication.
+
+```tsx
+// pages/github/authorizing.tsx
+
+import { useGithubAuthRedirect } from 'react-tinacms-github'
+
+// Our GitHub app redirects back to this page with auth code
+export default function Authorizing() {
+  // Let the main app know, that we received an auth code from the GitHub redirect
+  useGithubAuthRedirect()
+
+  return <h2>Authorizing with GitHub, Please wait...</h2>
+}
+```
+
+## Setup the GitHub OAuth App
+
+In GitHub, within your Account Settings, click <a href="https://github.com/settings/developers" target="_blank">Oauth Apps</a> under Developer Settings. Go ahead and create a "New Oauth App".
+
+For the **Authorization callback URL**, enter the url for the "/github/authorizing" page that you created above (e.g https://your-url/github/authorizing). Fill out the other fields with your custom values.
+
+> _Note:_ If you are **testing your app locally**, you may need a separate development GitHub app (with a localhost redirect), and a production GitHub app.
+
+Next we will add the **Client ID** and **Client Secret** to your environment variables.
+
+<!-- TODO: add ## Developing Locally section at this point? to explain where the content is sourcing from, etc. -->
+
+### Setting Environment Variables
```

Comment:
After this step it's possible to authenticate. It might be nice to point this out and get them to do so.